commit f20ad3ec8222ffa5168eae8931edcc3f0ac87d08
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Sat Jan 26 02:45:58 2019 +0800

    fix: crash when running lightdm-deepin-greeter
    
    The return value of createFrame was used and added to m_frames, so this
    function must return a valid frame instead of nothing.
    
    This results in an immediate crash in Arch:
    https://github.com/linuxdeepin/developer-center/issues/843
    
    Change-Id: I9941894c72a2c6fbd7c5fc45cd1299e23b1a6b2d

diff --git a/lightdm-deepin-greeter/main.cpp b/lightdm-deepin-greeter/main.cpp
index 95ca779..fd9914a 100644
--- a/lightdm-deepin-greeter/main.cpp
+++ b/lightdm-deepin-greeter/main.cpp
@@ -227,6 +227,7 @@ int main(int argc, char* argv[])
         QObject::connect(worker, &LockWorker::requestUpdateBackground, loginFrame, static_cast<void (LoginWindow::*)(const QString &)>(&LoginWindow::updateBackground));
         QObject::connect(loginFrame, &LoginWindow::destroyed, property_group, &PropertyGroup::removeObject);
         loginFrame->show();
+        return loginFrame;
     };
 
     MultiScreenManager multi_screen_manager;
